### PR TITLE
Fix switch camera and UI

### PR DESCRIPTION
**Pull request by Osman-Fatih-Cakir**

### CODE REVIEW

File: Editor/App.cpp [hunk 1]
```diff
@@ -1,3 +1,10 @@
+
+#include <filesystem>
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <memory>
```
Comment (afraidofdark) [inline Editor/App.cpp:7]: put an empty line to seperate system includes,
I recommend butting the correspondig headers at top of each cpp
App.h in this case

File: Editor/App.cpp [hunk 2]
```diff
@@ -849,11 +976,15 @@ namespace ToolKit
...
       SetCurrentScene(scene);
-      scene->Load(); // Make sure its loaded.
+      scene->Load();  // Make sure its loaded.
       scene->Init(false);
-      
+
       // Editor light retransfrom according to new camera
-      m_lightMaster->SetTransform(scene->GetCamera()->m_node->GetTransform(TransformationSpace::TS_LOCAL));
+      m_lightMaster->SetTransform
+      (
```
Comment (afraidofdark) [inline Editor/App.cpp:984]: fix this please

File: Editor/App.cpp [hunk 3]
```diff
@@ -868,13 +999,24 @@ namespace ToolKit
 
     void App::ApplyProjectSettings(bool setDefaults)
     {
-      if (CheckFile(ConcatPaths({ ResourcePath(), "Editor.settings" })) && !setDefaults)
+      if (CheckFile(ConcatPaths({ ResourcePath(), "Editor.settings" }))
```
Comment (afraidofdark) [inline Editor/App.cpp:1002]: fix this

File: Editor/App.cpp [hunk 4]
```diff
@@ -1011,7 +1164,8 @@ namespace ToolKit
         glColorMask(GL_FALSE, GL_FALSE, GL_FALSE, GL_FALSE);
 
         // webgl create problem with depth only drawing with textures.
-        static MaterialPtr solidMat = GetMaterialManager()->GetCopyOfSolidMaterial();
+        static MaterialPtr solidMat =
+          GetMaterialManager()->GetCopyOfSolidMaterial();
```
Comment (afraidofdark) [inline Editor/App.cpp:1168]: adjust your visual studio settings such that each new line start from the beggining of the previous line

File: Editor/EditorViewport.cpp [hunk 5]
```diff
@@ -226,11 +226,20 @@ namespace ToolKit
       *max = m_wndPos + m_wndContentAreaSize;
     }
 
-    void EditorViewport::SetCamera(Camera* cam)
+    Camera* EditorViewport::GetCamera() const
     {
-      Viewport::SetCamera(cam);
+      Camera* cam = static_cast<Camera*>
+        (
```
Comment (afraidofdark) [inline Editor/EditorViewport.cpp:232]: fix (
    
)

File: ToolKit/Viewport.cpp [hunk 6]
```diff
@@ -1,4 +1,7 @@
 #include "Viewport.h"
```
Comment (afraidofdark) [inline ToolKit/Viewport.cpp:1]: viewport.h goes to top

File: ToolKit/Viewport.cpp [hunk 7]
```diff
@@ -102,7 +108,12 @@ namespace ToolKit
     }
     else
     {
-      m_viewportImage = new RenderTarget((uint)m_width, (uint)m_height, settings);
+      m_viewportImage = new RenderTarget
```
Comment (afraidofdark) [inline ToolKit/Viewport.cpp:111]: does this pass linter ?

File: ToolKit/Viewport.cpp [hunk 8]
```diff
@@ -126,7 +137,10 @@ namespace ToolKit
     }
     else
     {
```
Comment (afraidofdark) [inline ToolKit/Viewport.cpp:139]: fix

File: ToolKit/Viewport.cpp (hunk 7 above)
Comment (Osman-Fatih-Cakir) [reply ToolKit/Viewport.cpp:111]: All files passes linter

File: Editor/EditorViewport.cpp [hunk 9]
```diff
@@ -226,11 +228,20 @@ namespace ToolKit
       *max = m_wndPos + m_wndContentAreaSize;
     }
 
-    void EditorViewport::SetCamera(Camera* cam)
+    Camera* EditorViewport::GetCamera() const
+    {
+      Camera* cam = static_cast<Camera*>
+      (
+        g_app->GetCurrentScene()->GetEntity(m_attachedCamera)
+      );
```
Comment (afraidofdark) [inline Editor/EditorViewport.cpp:236]: deleteCamera is unused revert function back to its original form

**Review by afraidofdark — CHANGES_REQUESTED** [review]
There mostly formating fixes that you need to do

**Pull request merged**